app/modals/receive-stx/receive-stx-modal.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Qr from 'qrcode.react';
+import { Text, Modal, Button, Flex, Box, useClipboard } from '@blockstack/ui';
+
+import { TxModalHeader, TxModalFooter } from '../transaction/transaction-modal-layout';
+import { homeActions, selectReceiveModalOpen } from '../../store/home/home.reducer';
+
+interface ReceiveStxModalProps {
+  address: string;
+}
+
+export const ReceiveStxModal: FC<ReceiveStxModalProps> = ({ address }) => {
+  const dispatch = useDispatch();
+  const modalOpen = useSelector(selectReceiveModalOpen);
+  const copyAddressToClipboard = useClipboard(address);
+  const [buttonText, setButtonText] = useState('Copy address');
+  const onCopyAddress = () => {
+    copyAddressToClipboard.onCopy();
+    setButtonText('Copied');
+    setTimeout(() => setButtonText('Copy address'), 800);
```

Comment:
This functionality is in the useClipboard hook -> https://github.com/blockstack/ux/blob/master/packages/ui/src/use-clipboard/index.tsx#L36-L40